sources/iTermApplicationDelegate.m

```diff
@@ -2063,7 +2063,17 @@ - (IBAction)loadRecording:(id)sender {
 #pragma mark - Private
 
 - (void)updateProcessType {
-    [[iTermApplication sharedApplication] setIsUIElement:[iTermPreferences boolForKey:kPreferenceKeyUIElement]];
+    bool enableLSUI = [iTermPreferences boolForKey:kPreferenceKeyUIElement];
+
+    if ([iTermAdvancedSettingsModel automaticallyToggleLSUI]) {
+        bool onlyHotKeyWindowsOpen = true;
+        for (PseudoTerminal* term in [self terminals]) {
+            onlyHotKeyWindowsOpen &= [term isHotKeyWindow];
```

Comment:
A BOOL is a char, which means there are 255 nonzero values that are true. That makes bitwise and unsafe, unfortunately.

A better way to do this is:
```
const BOOL onlyHotKeyWindowsOpen = [self.terminals allWithBlock:^BOOL(PseudoTerminal *term) { return term.isHotKeyWindow; }];
```